test/Spec.hs

```diff
@@ -1,55 +1,80 @@
 
-data Foo = Foo | Bar Int | FooBar Int Text deriving (Generic, Typeable, Show)
 
-data Bar a b m = Bar1 (Maybe a) | Bar2 (Either a b) | Bar3 a
-  | Bar4 {
-      myMonadicResult :: m b
-  } deriving (Generic, Typeable, Show)
+allTests :: Spec
+allTests =
+  describe "buildBridge" $ do
+    it "tests with Int" $
+      let bst = buildBridge defaultBridge (mkTypeInfo (Proxy :: Proxy Int))
+          ti  = TypeInfo { _typePackage    = "purescript-prim"
+                       , _typeModule     = "Prim"
+                       , _typeName       = "Int"
+                       , _typeParameters = []}
+       in bst `shouldBe` ti
+    it "tests with custom type Foo" $
+      let bst = bridgeSumType (buildBridge defaultBridge) (mkSumType (Proxy :: Proxy Foo))
+          st = SumType
+                TypeInfo { _typePackage = "" , _typeModule = "TestData" , _typeName = "Foo" , _typeParameters = [] }
+                [ DataConstructor { _sigConstructor = "Foo" , _sigValues = Left [] }
+                , DataConstructor
+                  { _sigConstructor = "Bar"
+                  , _sigValues = Left [ TypeInfo { _typePackage = "purescript-prim" , _typeModule = "Prim" , _typeName = "Int" , _typeParameters = [] } ]
+                  }
+                , DataConstructor
+                  { _sigConstructor = "FooBar"
+                  , _sigValues = Left [ TypeInfo { _typePackage = "purescript-prim" , _typeModule = "Prim" , _typeName = "Int" , _typeParameters = [] }
+                                      , TypeInfo { _typePackage = "purescript-prim" , _typeModule = "Prim" , _typeName = "String" , _typeParameters = [] }
+                                      ]
+                  }
+                ]
+       in bst `shouldBe` st
+    it "tests the generation of a whole (dummy) module" $
+      let advanced = bridgeSumType (buildBridge defaultBridge) (mkSumType (Proxy :: Proxy (Bar A B M1)))
+          modules = sumTypeToModule advanced Map.empty
+          m = head . map moduleToText . Map.elems $ modules
+          txt = T.unlines [ "-- File auto generated by purescript-bridge! --"
+                          , "module TestData where"
+                          , ""
+                          , "import Data.Either (Either)"
+                          , "import Data.Maybe (Maybe)"
+                          , ""
+                          , "import Data.Generic (class Generic)"
+                          , ""
+                          , ""
+                          , "data Bar a b m ="
+                          , "    Bar1 (Maybe a)"
+                          , "  | Bar2 (Either a b)"
+                          , "  | Bar3 a"
+                          , "  | Bar4 {"
+                          , "      myMonadicResult :: m b"
+                          , "    }"
+                          , ""
+                          , "derive instance genericBar :: Generic Bar"
+                          , ""
+                          ]
```

Comment:
Der Test ist ein bisschen fragil, geht aber nicht wirklich besser, nehme ich an. hah - doch - aber deutlich aufwendiger: Man testet den generierten Code einfach in einem Test-Purescript projekt und schaut ob das kompiliert.